cadence-macros/src/state.rs

```diff
@@ -65,7 +65,7 @@ impl<T> SingletonHolder<T> {
     pub fn set(&self, val: T) {
         if self
             .state
-            .compare_exchange(UNSET, LOADING, Ordering::SeqCst, Ordering::SeqCst)
+            .compare_exchange(UNSET, LOADING, Ordering::AcqRel, Ordering::Relaxed)
```

Comment:
`Ordering::Acquire` might be enough (instead of `AcqRel`) as the release store is done below and that should establish happens-before relationship protecting `self.value` from accessing it by getting / setting it by multiple threads at once.